x-pack/legacy/plugins/security/public/views/management/roles_grid/components/roles_grid_page.tsx

```diff
@@ -218,6 +220,62 @@ class RolesGridPageUI extends Component<Props, State> {
           ) : null;
         },
       },
+      {
+        name: intl.formatMessage({
+          id: 'xpack.security.management.roles.actionsColumnName',
+          defaultMessage: 'Actions',
+        }),
+        width: '150px',
+        actions: [
+          {
+            available: (role: Role) => !isReadOnlyRole(role),
+            render: (role: Role) => {
+              const title = intl.formatMessage(
+                {
+                  id: 'xpack.security.management.roles.editRoleActionName',
+                  defaultMessage: `Edit {roleName}`,
+                },
+                {
+                  roleName: role.name,
+                }
+              );
+
+              return (
+                <EuiButtonIcon
+                  data-test-subj={`edit-role-action-${role.name}`}
+                  title={title}
+                  color={'primary'}
+                  iconType={'pencil'}
+                  href={getRoleManagementHref('edit', role.name)}
+                />
+              );
+            },
+          },
+          {
+            render: (role: Role) => {
+              const title = intl.formatMessage(
```

Comment:
I don't know how I feel about cloning reserved roles. We generally introduce reserved roles so that we have control over the privileges, so allowing them to be easily cloned might violate the benefit that we get from being able to add additional privileges without requiring user intervention.

Also, cloning the reserved roles which are associated with reserved Kibana privileges is going to cause problems both in the UI validation, and when submitting them to the API: https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/security/server/routes/api/external/roles/put.test.js#L421

Reply:
You make some good points here! I'll make it so only custom roles can be cloned